ArchiSteamFarm/Bot.cs

```diff
@@ -541,6 +541,11 @@ internal sealed class Bot : IDisposable {
 						}
 					}
 				}
+				else if (botName.StartsWith("R!", StringComparison.OrdinalIgnoreCase)) {
+					List<Bot> regexMatches = Bots.Where(kvp => Regex.Match(kvp.Key, botName, RegexOptions.IgnoreCase).Success).Select(kvp => kvp.Value).ToList();
```

Comment:
`ToList()` call is wasted performance since `UnionWith()` accepts `IEnumerable`. Instead just remove it and make `regexMatches` `IEnumerable`.